pkg/cli/admin/release/extract_tools.go

```diff
@@ -492,17 +492,16 @@ func (o *ExtractOptions) extractCommand(command string) error {
 
 		// copy the input to disk
 		if target.InjectReleaseImage {
-			releaseInfo := exactReleaseImage
 			var matched bool
 			errmsg := ""
 			switch target.Command {
-			case "oc":
-				releaseInfo = releaseName
-				errmsg = fmt.Sprintf("warning: Unable to inject release image info into %s, could not determine version", target.Command)
-			default:
+			case "openshift-install":
 				errmsg = fmt.Sprintf("warning: Unable to inject release image info into %s, installer will not be locked to the correct image", target.Command)
+				matched, err = copyAndReplaceReleaseInfo(w, r, 4*1024, exactReleaseImage)
+			default:
+				errmsg = fmt.Sprintf("warning: Unable to inject release image info into %s, could not determine version", target.Command)
+				matched, err = copyAndReplaceReleaseInfo(w, r, 4*1024, releaseName)
```

Comment:
Version and image are unrelated and should not be in the same code path.

Reply:
ok, will rework